### ISSUE TITLE
How to hard reset in current default branch

### ISSUE THREAD

**Issue by LovesAsuna** [gitbutlerapp/gitbutler#4791]
### Version

0.12.21

### Operating System

Mac OS X

### Distribution Method

dmg (Apple Silicon)

### Describe the issue

<img width="386" alt="image" src="https://github.com/user-attachments/assets/806bd464-1196-49fe-a3c4-9e073eec9629">
How do I force push (hard reset) after undoing in the same virtual branch?

**Comment by Byron** [comment]
I don't think that force-pushes from virtual branches are currently supported, but am also not entirely sure I am not missing something.
Maybe @krlvi could fill in the blanks.

**Comment by LovesAsuna** [comment]
Is there a good solution for this situation? It seems like I can only force push from traditional branches.

**Comment by H1ghBre4k3r** [comment]
Hey @LovesAsuna, does GitButler show you the classic `Force Push` button below the commits? 

Furthermore, did you explicitly disable force pushing for your repository within the settings?

**Comment by LovesAsuna** [comment]
For example, I created a file named "a" and a file named "b," and committed them with the messages "add file a" and "add file b," respectively. At this point, I no longer want the commit "add file b," so I want to perform a hard reset. Note that these two commits are on the same commit chain. The force push button you mentioned will appear in cases where I have edited the commit message or squashed commits, causing a conflict between the local and remote commits.

**Comment by estib-vega** [comment]
@LovesAsuna It's a good question. I don't see any reason not to allow this.
I'll take a look
